constellations/gemini-admin/src/utils/inputSelect.js

```diff
@@ -0,0 +1,105 @@
+"use strict";
+require("./inputText.sass");
+import React from "react";
+import Checkbox from "../../assets/checkmark_green.svg";
+
+/*
+ * Props for the InputSelect Component:
+ *
+ * value: Current selected option
+ *
+ * onChangeCallback: Function that is called when the selection changes.
+ *
+ * required: Whether or not the selection  is required (omit if false).
+ *
+ * hasNoDefault: If true, an extra option -- Select an option -- will be added as the default.
+ *                  It cannot be reselected after another option is chosen.
+ *
+ * label: A large gray label for the select, e.g. "ProgramID".
+ *
+ * description (optional): A more detailed description for what the selection is.
+ *
+ * options: A list of option objects:
+ *      [
+ *          { value: "ap_calc", displayName: "AP Calculus" }
+ *      ]
+ *  becomes
+ *      <option value="ap_calc">AP Calculus</option>
+ *  value should not be an empty string.
+ */
+export class InputSelect extends React.Component {
+    state = {
+        chosen: false,
+    };
+
+    onChangeSelect = (e) => {
+        this.setState({ chosen: true });
+        this.props.onChangeCallback(e);
+    };
+
+    renderDescription = (description, required, pass) => {
+        var formatDescription = <h4 className="hidden"></h4>;
+        if (!!description) {
+            let ending = required ? (
+                <span className={pass ? "" : "red"}>{" (required)"}</span>
+            ) : (
+                " (optional)"
+            );
+            formatDescription = (
+                <h4>
+                    {description} {ending}
+                </h4>
+            );
+        }
+        return formatDescription;
+    };
+
+    renderSelect = (options, value) => {
+        var optionElements = options.map((option, index) => (
+            <option key={index} value={option.value}>
+                {option.displayName}
+            </option>
+        ));
+
+        var defaultOption = this.props.hasNoDefault ? (
+            <option disabled selected value>
+                {" "}
+                -- Select an option --{" "}
```

Comment:
Hm this is odd formatting. Anyway we can just get this to be: `{" "}-- Select an option --{" "}` or use HTML code for a single space? Like `&nbsp;`?

Reply:
```
&nbsp;
-- Select an option --
&nbsp;
```

does that work? ^